org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/SynchronizeGradleBuildOperation.java

```diff
@@ -240,11 +249,74 @@ private IProject addNewEclipseProjectToWorkspace(EclipseProject project, SubMoni
     }
 
     private void uncoupleWorkspaceProjectFromGradle(IProject workspaceProject, SubMonitor monitor) {
-        monitor.setWorkRemaining(3);
+        monitor.setWorkRemaining(4);
         monitor.subTask(String.format("Uncouple workspace project %s from Gradle", workspaceProject.getName()));
         CorePlugin.workspaceOperations().refreshProject(workspaceProject, monitor.newChild(1, SubMonitor.SUPPRESS_ALL_LABELS));
+        unconfigureConfigurators(workspaceProject, monitor.newChild(1));
         CorePlugin.workspaceOperations().removeNature(workspaceProject, GradleProjectNature.ID, monitor.newChild(1, SubMonitor.SUPPRESS_ALL_LABELS));
         CorePlugin.modelPersistence().deleteModel(workspaceProject);
         CorePlugin.configurationManager().deleteProjectConfiguration(workspaceProject);
     }
+
+    private void initConfigurators(IProgressMonitor monitor) {
+        GradleBuild gradleBuild = GradleCore.getWorkspace().createBuild(this.buildConfig.toApiBuildConfiguration());
+        InitializationContext context = new InitializationContext() {
+
+            @Override
+            public GradleBuild getGradleBuild() {
+                return gradleBuild;
+            }
+        };
+
+        SubMonitor progress = SubMonitor.convert(monitor);
+        progress.setWorkRemaining(this.configurators.size());
+        for (ProjectConfigurator c : this.configurators) {
+            try {
+                c.init(context, progress.newChild(1));
+            } catch (Exception e) {
+                CorePlugin.logger().warn("Project configurator " + c.getClass().getName() + " failed to initialize" , e);
```

Comment:
Shouldn't we use the logger associated with the plugin that contributed the configurator? Otherwise bugs will be submitted to our bug tracker instead of theirs.